rfcs/2021-19-02-units-for-typography-and-layout.md

```diff
@@ -0,0 +1,615 @@
+
+Units are one of the core foundation principle and building block that will define how our overall designs i.e components, layouts shall be designed as we progress so that we can make our apps work on different platforms, different screen sizes and different pixel densities.
+
+### What use cases does it support?
+It'll define what units we'll be using for different styling properties in our design system. Below are the categories to which units will be applied
+1. Typography - font-size, line-height
+2. Layout - padding, margin, width, height
+
+### What is the expected outcome?
+We shall be able to answer following questions as a conclusion to this RFC
+1. What unit we'll be using for typography and layouts?
+2. How will we ensure accessibility in terms of readability and layout both?
+3. How will we use platform dependent units(px, rem, dp etc) in order to render the components on a particular platform in a generic way without polluting our design system components.
+4. How/where we will store the unit as design tokens?
+
+# Lesser known facts
+1. `1rem` = `16px` - default in all the browsers unless the default font-size of the browser is set to anything else explicitly.
+2. `padding: 1rem` - rems always takes base value as font-size of the browser regardless of the property they are used on. In this example 1rem = 16px so `padding: 16px`.
+3. `1dp = 1px` at `360x640` and `160ppi`(a.k.a `@1x`), therefore `1dp = 3px` at `1080x1920` and `440ppi`(a.k.a `@3x`).
+
+# Detailed Design
+For anything to be rendered on the screen it needs to have a space on the screen and that space needs to have a measurable unit for the rendering engine to understand how to organise things on the screen.
+
+The styling properties that require units are:
+1. Typography
+   * `font-size`
+   * `line-height`
+2. Layout
+   1. Space
+      * `padding` - top, right, bottom, left
+      * `margin` - top, right, bottom, left
+   2. Size
+      * `height` - height, min-height, max-height
+      * `width` - width, min-width, max-width
+
+## Screen resolutions and densities
+One of the main goal for layouts and content on the screen is that they should adapt to different screen resolutions and densities and things should still visually appear the same. Now what are some different factors that affect this?
+
+### Aspect Ratio
+Aspect ratio is basically `width/height` of a screen i.e find ratios with the greatest common divisor
+
+Examples:
+
+800x600 = 4:3
+
+1024x768 = 4:3
+
+1920x1200 = 16:10
+
+3840x2160 = 16:9
+### PPI
+PPI is short for pixels per inch which says the number of pixels that can be accommodated in 1 sq inch on the screen. 
+
+PPI plays a signficant role of how things are physically rendered on the screen
+
+How is PPI calculated? **`area of screen/resolution`**
+
+Example:
+
+_Area of **20"** Screen: **20.5x12.5=256 sq.in**_
+
+_Screen Resolution: **1920*1080 = 2073600 (21600 inches)**_
+
+_PPI: **21600/256 = 84ppi**_
```

Comment:
The math still seems confusing to me. Might be good idea to just link to a PPI calculator here instead of deriving things. Also the numbers don't seem to match with some online calculators I tried to understand the calculations, for example [this one](https://www.sven.de/dpi/).

Reply:
Sure I've updated it.